### PR TITLE
Targetarch fix docker

### PR DESCRIPTION
**Pull request by thomas-fossati**
This PR contains a bunch of fixes on top of Ionut's `targetarch` branch that
* explicitly state the startup dependencies among the various containers
* create explicit `/etc/hosts` entries for communication between the veraison services

This should work around a race condition in Veraison services startup.

### CODE REVIEW

**Review by paulhowardarm — APPROVED** [review]
Looks good - and thanks so much for your perseverance on the weird dependency issue!

File: demo/attester/attester.Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM ubuntu:20.04
 
+ARG TARGETARCH
```
Comment (ionut-arm) [inline demo/attester/attester.Dockerfile:3]: Hmm, this was already fixed in [my PR](https://github.com/CCC-Attestation/attested-tls-poc/pull/46/files). I get the feeling that our merge strategy isn't ideal :(

File: demo/build/veraison.mk [hunk 2]
```diff
@@ -1,10 +1,17 @@
-
 VERAISON_ROOT := .veraison-services
 
+# The command line can override VERAISON_BRANCH to point to a branch other than
+# main
+VERAISON_BRANCH ?= main
```
Comment (ionut-arm) [inline demo/build/veraison.mk:5]: Could we tie this to a particular commit? Otherwise it's not strictly reproducible.

**Review by ionut-arm — COMMENTED** [review]
Thanks for the fixes! Got a couple of comments

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

File: demo/build/veraison.mk (hunk 2 above)
Comment (thomas-fossati) [reply demo/build/veraison.mk:5]: true dat.  I'd like to defer this though cos I sense there may be multiple ways to do this (e.g., using hooks from the veraison docker machinery) which I'd like to survey first.

**Comment by thomas-fossati** [comment]
> Looks good - and thanks so much for your perseverance on the weird dependency issue!

thank you for the moral support :-)

File: demo/attester/attester.Dockerfile (hunk 1 above)
Comment (thomas-fossati) [reply demo/attester/attester.Dockerfile]: rebase done

**Pull request merged**